test/integration/auth-methods/jwt.test.js

```diff
@@ -0,0 +1,1664 @@
+        const { alg, signKey, publicKey } = algorithmsSupported[key];
+
+        describe(`with valid header object for ${alg} algorithm`, function () {
+            before(function (done) {
+                const runOptions = {
+                    collection: {
+                        item: {
+                            request: {
+                                url: 'https://postman-echo.com/headers',
+                                auth: {
+                                    type: 'jwt',
+                                    jwt: {
+                                        algorithm: alg,
+                                        header: {
+                                            alg: alg,
+                                            typ: 'JWS',
+                                            cty: 'JWT',
+                                            crit: ['JWS'],
+                                            kid: '12345',
+                                            jku: '{{jku}}'
+                                        },
+                                        payload: { test: 123 },
+                                        secret: signKey,
+                                        privateKey: signKey,
+                                        tokenAddTo: HEADER
+                                    }
+                                }
+                            }
+                        }
+                    },
+                    environment: {
+                        values: [
+                            {
+                                key: 'jku',
+                                value: 'test-jku'
+                            }
+                        ]
+                    }
+                };
+
+                this.run(runOptions, function (err, results) {
+                    testrun = results;
+                    done(err);
+                });
+            });
+
+            it('should completed the run', function () {
+                expect(testrun).to.be.ok;
+                expect(testrun).to.nested.include({
+                    'done.calledOnce': true,
+                    'start.calledOnce': true,
+                    'request.calledOnce': true
+                });
+            });
+
+            it('should add Authorization header', function () {
+                const headers = [],
+                    request = testrun.request.firstCall.args[3],
+                    response = testrun.request.firstCall.args[2];
+
+                let jwtToken;
+
+                request.headers.members.forEach(function (header) {
+                    if (header.key === 'Authorization') {
+                        jwtToken = header.value.split('Bearer ')[1];
+                    }
+                    headers.push(header.key);
+                });
+
+                expect(request.headers.members).to.include.deep.members([
+                    new Header({ key: 'Authorization', value: `Bearer ${jwtToken}`, system: true })
```

Comment:
What is this test doing? It's taking out the token from the request headers and test it to itself?

Reply:
Yes @coditva  the jwtToken is generated and added to request, we are taking it out to verify the signature & payload to make sure it is correctly generated

Reply:
This `expect` seems weird because we took out the token from the headers, then compared the headers with the value we got. Something like:
```javascript
token = val;

expect(val).to.equal(token);
```

This `expect` block will never fail, no?

Reply:
Yes @coditva agree that expect block to again is unnecessary will remove it and keep the verifying part alone